kp_test.go

```diff
@@ -2213,3 +2214,172 @@ func TestCancel_DualAuthDelete(t *testing.T) {
+
+}
+
+func TestGetKeyWithAlias(t *testing.T) {
+	defer gock.Off()
+	keyResponse := []byte(`{
+		"metadata": {
+			"collectionTotal": 1,
+			"collectionType": "application/vnd.ibm.kms.key+json"
+		},
+		"resources": [
+			{
+				"algorithmType": "AES",
+				"aliases": [
+					"inumu",
+					"domainblvd"
+				],
+				"createdBy": "xyz",
+				"creationDate": "2020-11-06T22:40:44Z",
+				"crn": "crn:v1:staging:public:kms:us-south:a/07214fad6bb9305647dc3ebe3244b781:43d105f4-84a0-41c8-8950-4b7180ffa213:key:f4877b07-40b4-4307-a306-eb98eeeea590",
+				"deleted": false,
+				"description": "a testing thing",
+				"dualAuthDelete": {
+					"enabled": false
+				},
+				"extractable": true,
+				"id": "f4877b07-40b4-4307-a306-eb98eeeea590",
+				"payload": "1/9ajNlN5Vu9s2rR7mLhOVR6aGn8+gbtgj03PwfFUaM=",
+				"lastUpdateDate": "2020-11-06T22:40:44Z",
+				"name": "test secret",
+				"state": 1,
+				"type": "application/vnd.ibm.kms.key+json"
+			}
+		]
+	}`)
+	alias := "domainblvd"
+
+	gock.New("http://example.com").
+		Get("/api/v2/keys/" + alias).
+		Reply(200).
+		Body(bytes.NewReader(keyResponse))
+
+	c, _, err := NewTestClient(t, nil)
+	gock.InterceptClient(&c.HttpClient)
+	defer gock.RestoreClient(&c.HttpClient)
+	c.tokenSource = &FakeTokenSource{}
+
+	key, err := c.GetKey(context.Background(), alias)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, key)
+	assert.NotEqual(t, key.Payload, "")
+	assert.True(t, slice.ContainsString(key.Aliases, alias))
+
+	assert.True(t, gock.IsDone(), "Expected HTTP requests not called")
+}
+
+func TestGetKeyMetadataWithAlias(t *testing.T) {
+	defer gock.Off()
+	keyResponse := []byte(`{
+		"metadata": {
+			"collectionTotal": 1,
+			"collectionType": "application/vnd.ibm.kms.key+json"
+		},
+		"resources": [
+			{
+				"algorithmType": "AES",
+				"aliases": [
+					"inumu",
+					"domainblvd"
+				],
+				"createdBy": "xyz",
+				"creationDate": "2020-11-06T22:40:44Z",
+				"crn": "crn:v1:staging:public:kms:us-south:a/07214fad6bb9305647dc3ebe3244b781:43d105f4-84a0-41c8-8950-4b7180ffa213:key:f4877b07-40b4-4307-a306-eb98eeeea590",
+				"deleted": false,
+				"description": "a testing thing",
+				"dualAuthDelete": {
+					"enabled": false
+				},
+				"extractable": true,
+				"id": "f4877b07-40b4-4307-a306-eb98eeeea590",
+				"lastUpdateDate": "2020-11-06T22:40:44Z",
+				"name": "test secret",
+				"state": 1,
+				"type": "application/vnd.ibm.kms.key+json"
+			}
+		]
+	}`)
+	alias := "domainblvd"
+
+	gock.New("http://example.com").
+		Get("/api/v2/keys/" + alias + "/metadata").
+		Reply(200).
+		Body(bytes.NewReader(keyResponse))
+
+	c, _, err := NewTestClient(t, nil)
+	gock.InterceptClient(&c.HttpClient)
+	defer gock.RestoreClient(&c.HttpClient)
+	c.tokenSource = &FakeTokenSource{}
+
+	key, err := c.GetKeyMetadata(context.Background(), alias)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, key)
+	assert.Equal(t, key.Payload, "")
+	assert.True(t, slice.ContainsString(key.Aliases, alias))
```

Comment:
Lets try to use https://godoc.org/github.com/stretchr/testify/assert#Contains instead of a new third party library.

We can still use the library if the testify method does not work as needed.